packages/dds/tree/src/test/shared-tree/fuzz/undoRedo.fuzz.spec.ts

```diff
@@ -69,232 +70,170 @@ describe("Fuzz - undo/redo", () => {
 			Operation,
 			DDSFuzzTestState<SharedTreeTestFactory>
 		> = {
-			workloadName: "SharedTree",
-			factory: new SharedTreeTestFactory(onCreate),
+			workloadName: "revert sequenced commits last-to-first",
+			factory: new SharedTreeTestFactory(() => {}),
 			generatorFactory,
 			reducer: fuzzReducer,
-			validateConsistency: () => {},
+			validateConsistency: validateTreeConsistency,
 		};
 		const emitter = new TypedEventEmitter<DDSFuzzHarnessEvents>();
-		emitter.on("testStart", (initialState: UndoRedoFuzzTestState) => {
-			const tree = viewFromState(initialState).checkout;
-			initialState.initialTreeState = toJsonableTree(tree);
-			initialState.anchors = [];
-			for (const client of initialState.clients) {
-				const view = viewFromState(initialState, client)
-					.checkout as RevertibleSharedTreeView;
-				const { undoStack, redoStack, unsubscribe } = createTestUndoRedoStacks(view.events);
-				view.undoStack = undoStack;
-				view.redoStack = redoStack;
-				view.unsubscribe = unsubscribe;
-				initialState.anchors.push(createAnchors(view));
+		emitter.on("testStart", (state: UndoRedoFuzzTestState) => {
+			const tree = viewFromState(state, state.clients[0], populatedInitialState).checkout;
+			state.containerRuntimeFactory.processAllMessages();
+			state.initialTreeState = toJsonableTree(tree);
+			const undoStack: Revertible[] = [];
+			const redoStack: Revertible[] = [];
+			state.undoStack = undoStack;
+			state.redoStack = redoStack;
+			state.anchors = [];
+			for (const client of state.clients) {
+				const checkout = viewFromState(state, client).checkout;
+				checkout.events.on("commitApplied", (commit, getRevertible) => {
```

Comment:
should we unsubscribe each client when the test ends too or does it not matter because it's a test?

Reply:
For a single test it wouldn't worry me too much, but since this is per seed, it may add up. I'll fix it.